### ISSUE TITLE
Static methods

### ISSUE THREAD

**Issue by vorov2** [vorov2/dyalect#27]
**Is your feature request related to a problem? Please describe.**
Consider implementing static methods, which could be used for example for type construction.

**Describe the solution you'd like**
A method is a function which implementation is a part of a value prototype (which we call just a "type"), e.g.:

```swift
var str = "foo"
str.len() //len is a method of a String type
```

A method is accessed through an object instance. That is Ok. However sometimes we might need to invoke a certain method on the type itself, for example, in order to construct an instance:

```swift
var str = String.concat("foo", "bar")
```

Also there should be a way to define such methods in Dy. E.g.:

```swift
func String.len() { //regular method
}

static func String.concat(args...) { //static method
}
```
The syntax of statics above is for illustrative purposes only. The final syntax is yet to be decided.

**Describe alternatives you've considered**
Nothing yet, but there is some room for decision making.

**Comment by vorov2** [comment]
The first implementation would be available in 0.4.0.

**Comment by vorov2** [comment]
In master
